docs/research/prototypes/diagram-bake-off/depth_code2flow.py

```
@@ -0,0 +1,150 @@
+from pathlib import Path
+
+PROTO_ROOT = Path(__file__).resolve().parent
+OUT_DIR = PROTO_ROOT / "out"
+
+#: Populated by `depth_edges` as a side effect — see emit.py's driver.
+STATS: dict = {}
+
+
+def _run_code2flow(repo_root: Path, out_json: Path) -> tuple[dict, list[str]]:
+    target = repo_root / "python" / "src" / "kb_setup"
+    proc = subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)